source/Sashimi.AzureServiceFabric/Endpoints/AzureServiceFabricClusterEndpoint.cs

```diff
@@ -54,5 +55,7 @@ public override IEnumerable<(string id, DocumentType documentType)> GetRelatedDo
             if (!string.IsNullOrEmpty(DefaultWorkerPoolId))
                 yield return (DefaultWorkerPoolId, DocumentType.WorkerPool);
         }
+        
+        public IEnumerable<string> ClientCertificateIds => new[] { ClientCertVariable }.Where(c => !string.IsNullOrEmpty(c)).Cast<string>();
```

Comment:
had to do a Cast as C# 8 compiler check was complaining about string? not usable as string

Reply:
Yep.. we're going to encounter a bunch of this kind of casting/null-checking fun until the nullability changes roll all the way through the codebase. Thankfully, R# should start highlighting the redundancies for us once they _are_ redundant.

Reply:
Is the intent to get a static answer back or to re-evaluate the answer every time the returned value is iterated? My preference is to return `string[]` to ensure this is a once off `.Where(...).Cast<string>()`. I.e. you'll getting the `multiple enumerations in the calling code warning from this atm if you use the value more than once in the calling code.